### PR TITLE
Fixes to get successful transpilation and test runs on TS 2.1

### PR DESCRIPTION
**Pull request by rorticus**
Fixing some code in `Task` that causes incompatibilities with TypeScript 2.1

### CODE REVIEW

File: src/async/Task.ts [hunk 1]
```diff
@@ -80,35 +80,47 @@ export default class Task<T> extends ExtensiblePromise<T> {
...
-		this._state = State.Pending;
-
-		this.children = [];
-		this.canceler = () => {
-			if (canceler) {
-				canceler();
-			}
-			this._cancel();
-		};
+		} catch (reason) {
```
Comment (kitsonk) [inline src/async/Task.ts]:
``` ts
}
catch (reason) {
```

**Review by kitsonk — COMMENTED** [review]
One minor style comment

**Comment by kitsonk** [comment]
Refs #238

File: src/async/Task.ts (hunk 1 above)
Comment (rorticus) [reply src/async/Task.ts]: Good `catch`. Ha! Get it!?

**Pull request merged**